recipes/verilator/all/conanfile.py

```diff
@@ -0,0 +1,225 @@
+        else:
+            self.build_requires("flex/2.6.4")
+            self.build_requires("bison/3.5.3")
+
+    def requirements(self):
+        if self.settings.os == "Windows":
+            self.requires("strawberryperl/5.30.0.1")
+        if self.settings.compiler == "Visual Studio":
+            self.requires("dirent/1.23.2", private=True)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("verilator-{}".format(self.version), self._source_subfolder)
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            build_env = {
+                "CC": "{} cl -nologo".format(tools.unix_path(self.deps_user_info["automake"].compile)),
+                "CXX": "{} cl -nologo".format(tools.unix_path(self.deps_user_info["automake"].compile)),
+                "AR": "{} lib".format(tools.unix_path(self.deps_user_info["automake"].ar_lib)),
+            }
+            with tools.vcvars(self.settings):
+                with tools.environment_append(build_env):
+                    yield
+        else:
+            yield
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+        self._autotools.libs = []
+        self._autotools.library_paths = []
+        if self.settings.compiler == "Visual Studio":
+            self._autotools.cxx_flags.append("-EHsc")
+            self._autotools.defines.append("YY_NO_UNISTD_H")
+            self._autotools.flags.append("-FS")
+        conf_args = [
+            "--datarootdir={}/bin/share".format(tools.unix_path(self.package_folder)),
+        ]
+        self._autotools.configure(args=conf_args, configure_dir=os.path.join(self.build_folder, self._source_subfolder))
+        return self._autotools
+
+    @property
+    def _make_args(self):
+        args = [
+            "CFG_WITH_DEFENV=false",
+            "SRC_TARGET={}".format("dbg" if self.settings.build_type == "Debug" else "opt"),
+        ]
+        if self.settings.build_type == "Debug":
+            args.append("DEBUG=1")
+        if self.settings.compiler == "Visual Studio":
+            args.append("PROGLINK={}".format(tools.unix_path(os.path.join(self.build_folder, "msvc_link.sh"))))
+        return args
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+        try:
+            os.unlink(os.path.join(self._source_subfolder, "src", "config_build.h"))
+        except FileNotFoundError:
+            pass
+
+        if self.settings.compiler == "Visual Studio":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "src", "Makefile_obj.in"),
+                                  "${LINK}", "${PROGLINK}")
+
+        tools.save("msvc_link.sh", textwrap.dedent("""\
```

Comment:
it could be a separated file

Reply:
I moved it to a patch.

Can you please take a look at the remaining blocker concerning builddirs and build_modules?

Reply:
Scrap that, I will push a fix to the hooks repo.

Reply:
@uilianries 
See https://github.com/conan-io/hooks/pull/200

Reply:
The tests are failing due to a recent conan addition...

Reply:
@madebr thanks for providing that fix!